### PR TITLE
Coverity Fixes for issue #12531

### PR DESCRIPTION
**Pull request by ashman-p**
Fixes #12531 on master branch.
Address comments from 
https://github.com/openssl/openssl/pull/12531

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/x509/pcy_data.c [hunk 1]
```diff
@@ -55,6 +55,7 @@ X509_POLICY_DATA *policy_data_new(POLICYINFO *policy,
     ret = OPENSSL_zalloc(sizeof(*ret));
     if (ret == NULL) {
         X509V3err(X509V3_F_POLICY_DATA_NEW, ERR_R_MALLOC_FAILURE);
+        ASN1_OBJECT_free(id);
```
Comment (slontis) [inline crypto/x509/pcy_data.c:58]: Small nit - maybe move this line up so it looks similar to the code below it.

File: crypto/asn1/x_algor.c [hunk 2]
```diff
@@ -110,13 +110,18 @@ int X509_ALGOR_copy(X509_ALGOR *dest, const X509_ALGOR *src)
         if ((dest->algorithm = OBJ_dup(src->algorithm)) == NULL)
 	    return 0;
 
-    if (src->parameter)
+    if (src->parameter) {
```
Comment (slontis) [inline crypto/asn1/x_algor.c:113]: May as well fix up this code while you are here..

if (src->parameter != NULL)

File: crypto/asn1/x_algor.c [hunk 3]
```diff
@@ -110,13 +110,18 @@ int X509_ALGOR_copy(X509_ALGOR *dest, const X509_ALGOR *src)
...
+        dest->parameter = ASN1_TYPE_new();
+        if (dest->parameter == NULL)
+            return 0;
+
         /* Assuming this is also correct for a BOOL.
```
Comment (slontis) [inline crypto/asn1/x_algor.c:118]: And here 

/*
 * Assuming

File: crypto/x509/pcy_data.c (hunk 1 above)
Comment (ashman-p) [reply crypto/x509/pcy_data.c:58]: Updated.

File: crypto/asn1/x_algor.c (hunk 2 above)
Comment (ashman-p) [reply crypto/asn1/x_algor.c:113]: Updated.

**Comment by ashman-p** [comment]
No worries at all.

On Jul 29, 2020, at 9:18 PM, Shane <notifications@github.com<mailto:notifications@github.com>> wrote:



@slontis approved this pull request.

Thanks for your patience..

**Comment by slontis** [comment]
Thanks for fixing. Merged to master.

**Pull request closed without merging**